### PR TITLE
gh-93. Move Redux auth logic to auth hook

### PR DESCRIPTION
**Pull request by yzhbankov**
close #gh-93

### CODE REVIEW

File: apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-import {selectLoginState, selectAuthError, selectAuthToken} from '../selectors'
 import {useAuth} from '../hooks'
 
 export function LoginPage() {
   const history = useHistory()
-  const isLogging = useSelector(selectLoginState)
-  const loginError = useSelector(selectAuthError)
-  const token = useSelector(selectAuthToken)
-  const auth = useAuth()
+  const {login, isLogging, loginError, token} = useAuth()
```
Comment (nesterone) [inline apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx]: 'logingError' -> error
isLoggin -> hasToken (or something more explanatory)

File: apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
   useEffect(() => {
     if (!isLogging && !loginError && token) {
       history.push('/home')
```
Comment (nesterone) [inline apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx]: What about using `<Redirect>` components instead?

**Review by nesterone — COMMENTED** [review]
What about navbar ? It also makes sence to switch to `useAuth`

File: apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx (hunk 1 above)
Comment (yzhbankov) [reply apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx]: done

File: apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx (hunk 2 above)
Comment (yzhbankov) [reply apps/web-portal-client/lib/features/auth/containers/LoginPage.jsx]: done

File: apps/web-portal-client/lib/features/auth/hooks/auth.jsx [hunk 3]
```diff
@@ -19,5 +20,11 @@ export function useAuth() {
           dispatch(error(err))
         })
     },
+    logout: () => {
+      dispatch(logout())
+    },
+    inProcess: useSelector(selectLoginState),
```
Comment (nesterone) [inline apps/web-portal-client/lib/features/auth/hooks/auth.jsx]: inProgress -> isLogginInProgress

File: apps/web-portal-client/lib/features/auth/hooks/auth.jsx [hunk 4]
```diff
@@ -19,5 +20,11 @@ export function useAuth() {
...
+    error: useSelector(selectAuthError),
+    loggedIn: useSelector(selectAuthToken),
```
Comment (nesterone) [inline apps/web-portal-client/lib/features/auth/hooks/auth.jsx]: loggedIn -> wasLoggedIn: Boolean(useSelector(selectAuthToken))
or hasToken: Boolean(useSelector(selectAuthToken))

File: apps/web-portal-client/lib/features/auth/hooks/auth.jsx [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import {useContext} from 'react'
-import {useDispatch} from 'react-redux'
-import {error, logging, login} from '../slices'
+import {useDispatch, useSelector} from 'react-redux'
+import {error, logging, login, logout} from '../slices'
 import {AppContext} from '../../../context'
+import {selectAuthError, selectAuthToken, selectLoginState} from '../selectors'
 
 export function useAuth() {
   const dispatch = useDispatch()
```
Comment (nesterone) [inline apps/web-portal-client/lib/features/auth/hooks/auth.jsx]: Looks like we have to wrap `dispatch` with `useCallback`

https://react-redux.js.org/next/api/hooks#examples

**Review by nesterone — CHANGES_REQUESTED** [review]

File: apps/web-portal-client/lib/features/auth/hooks/auth.jsx [hunk 6]
```diff
@@ -19,5 +20,11 @@ export function useAuth() {
           dispatch(error(err))
         })
     },
+    logout: () => {
+      dispatch(logout())
```
Comment (nesterone) [inline apps/web-portal-client/lib/features/auth/hooks/auth.jsx]: Do we have an API method to do logout?

**Review by nesterone — CHANGES_REQUESTED** [review]

File: apps/web-portal-client/lib/features/auth/hooks/auth.jsx (hunk 6 above)
Comment (yzhbankov) [reply apps/web-portal-client/lib/features/auth/hooks/auth.jsx]: No, we have not. It is not clear for now what auth mechanism we will use. I can add some plug

Comment (nesterone) [reply apps/web-portal-client/lib/features/auth/hooks/auth.jsx]: so extracting to separate task

**Comment by nesterone** [comment]
Almost forget,

`../hooks/auth.jsx` -> `../hooks/useAuth.jsx`

**Review by nesterone — APPROVED** [review]

**Pull request merged**